pygmt/datasets/samples.py

```diff
@@ -1,13 +1,100 @@
 """
 Functions to load sample data.
 """
+import warnings
+
 import pandas as pd
+from pygmt.exceptions import GMTInvalidInput
 from pygmt.src import which
 
 
-def load_japan_quakes():
+def list_sample_data():
+    """
+    Report datasets available for tests and documentation examples.
+
+    Returns
+    -------
+    output : dict
+        Names and short descriptions of available sample datasets.
+
+    See Also
+    --------
+    load_sample_data : Load an example dataset from the GMT server.
+    """
+    names = {
+        "japan_quakes": "Table of earthquakes around Japan from NOAA NGDC database",
+        "ocean_ridge_points": "Table of ocean ridge points for the entire world",
+        "bathymetry": "Table of ship bathymetric observations off Baja California",
+        "usgs_quakes": "Table of global earthquakes from the USGS",
+        "fractures": "Table of hypothetical fracture lengths and azimuths",
+        "hotspots": "Table of locations, names, and symbol sizes of hotpots from "
+        " Mueller et al., 1993",
+        "mars_shape": "Table of topographic signature of the hemispheric dichotomy of "
+        " Mars from Smith and Zuber (1996)",
+    }
+    return names
+
+
+def load_sample_data(name):
+    """
+    Load an example dataset from the GMT server.
+
+    The data are downloaded to a cache directory (usually ``~/.gmt/cache``) the
+    first time you invoke this function. Afterwards, it will load the data from
+    the cache. So you'll need an internet connection the first time around.
+
+    Parameters
+    ----------
+    name : str
+        Name of the dataset to load.
+
+    Returns
+    -------
+    output : pandas.DataFrame or xarray.DataArray
+        Sample dataset loaded as a pandas.DataFrame for tabular data or
+        xarray.DataArray for raster data
+
+    See Also
+    --------
+    list_sample_data : Report datasets available for tests and documentation
+        examples.
+    """
+    names = list_sample_data()
+    if name not in names:
+        raise GMTInvalidInput(f"Invalid dataset name '{name}'")
+
```

Comment:
Can we simply have a dict mapping a name to a load function, like this:
```
load_func = {
    "japan_quakes": load_japan_quakes,
    "ocean_ridge_points": load_ocean_ridge_points,
}

data = load_func[name](supress_warning=True)
```
I didn't test it so I don't know if it works or not.

Reply:
It works. I think it is simpler, thanks.